lib/extensions/ar_slug.rb

```diff
@@ -0,0 +1,13 @@
+module ArSlug
+  extend ActiveSupport::Concern
+
+  included do
+    virtual_column :href_slug, :type => :string
```

Comment:
I feel like we should namespace ourselves here and call it `api_href_slug` and the module itself `ArApiSlug` or `ArApiHrefSlug` or something like that, because we get slugs in the providers world for other purposes and it can get confusing.  e.g. ManageIQ::Providers::Redhat::Instance#href_slug.  Is that the ManageIQ href slug or is that the native RHV href slug?